### ISSUE TITLE
[QUESTION] Docker data-only container file permissions

### ISSUE THREAD

**Issue by oleynikd** [docker/docker#15828]
I've asked this questions on stackoverflows, #docker irc chat and couple of forums but still no solution. Please help!

I'm developing PHP app and I'm stuck with Docker volumes. I've tried to create a separate data-only container for my PHP app but couldn't make it work because of permission issues... I've googled and read all I could and the most close to working solution [is described here][1]

But it's a bit old and I couldn't make it work too.

I've created a repo with a simple test code:
https://github.com/oleynikd/docker-volumes-permissions/

This simple project uses docker-compose to run 2 containers:

 1. For php fpm
 2. For nginx

The php code is mapped to php container and lives in `/src` directory.

The main problem is that PHP has no rights to write to code's directory because it runs as **www-data** user but code's directory belongs to user with id **1000** and group **staff**.

After running docker-compose up and visiting /index.php you'll see the warning and `ls -lah` output that shows the permission issue. Here's the screenshot:[![enter image description here][2]][2]

I've tried to fix this by adding `RUN mkdir -p /src && chown -R www-data:www-data /src` to php Dockerfile but that didn't helped.

So questions are:

1. **Why the owner and the group of /src is 1000:staff?**
2. **How to fix this?**

I'm sure the solution is simple but I can't find it.
Please help!

P.S. Feel free to contribute to repo if you know how to fix this issue.

  [1]: http://stackoverflow.com/a/27021154/983914
  [2]: http://i.stack.imgur.com/cPCrZ.png

**Comment by oleynikd** [comment]
It's not a bug, it's just a help request...

**Comment by monitorjbl** [comment]
The container doesn't have a record of the users/groups your files refer to. You'll see the same phenomenon anytime you mount volumes into the container. To get around this, you could set up users/groups inside the container with matching UID/GIDs to the files outside the container.

**Comment by thaJeztah** [comment]
Afaik, this is a limitation of the VirtualBox shares used by Boot2Docker to share host directories with the Boot2Docker virtual machine. More information (and some workarounds) can be found here; https://github.com/boot2docker/boot2docker/issues/581#issuecomment-60314129

I'm going to close this issue, because it's not a bug, but feel free to comment here after I've closed this issue.

**Comment by lcrea** [comment]
@oleynikd 
Following your example, I personally solved the same problem with this simple workaround:

1. I copied locally the `/etc/apache2/apache2.conf` from a running container.
2. I changed the user group to **staff** (I left user as was).
3. I created a `Dockerfile` and added this lines:

```
FROM php:5-apache
COPY apache2.conf /etc/apache2/apache2.conf

RUN useradd -g staff -M -r -s /usr/bin/nologin -u 1000 denis \
   &&  chown -R www-data:staff /src
```

4. Finally, I built a new image.

Doing like this, you set apache server to run in the same group of the VirtualBox shared dir and, adding the user, you give it the write permission.
Obviously, it's not intended for live production, but it speeds up your development process giving you the ability to live preview your changes.

I really hope this could help you 😉
Bye!
